tests/devicetree_test.py

```diff
@@ -57,15 +58,39 @@ def test_resolve_device(self):
 
         self.assertEqual(dt.resolve_device(dev3.name), dev3)
 
+    def test_device_name(self):
+        # check that devicetree.names property contains all device's names
+        tree = DeviceTree()
+
+        dev_names = ["sda", "sdb", "sdc"]
+
+        for dev_name in dev_names:
+            dev = DiskDevice(dev_name)
+            tree._add_device(dev)
+            self.assertTrue(dev in tree.devices)
+
+        # frobnicate a bit with the hidden status of the devices:
+        # * hide sda
+        # * hide and unhide again sdb
+        # * leave sdc unchanged
+        tree.hide(tree.get_device_by_name("sda"))
+        tree.hide(tree.get_device_by_name("sdb"))
+
+        tree.unhide(tree.get_device_by_name("sdb", hidden=True))
+
+        # some lvs names may be already present in the system
+        lv_info = list(lvs_info.cache.keys())
```

Comment:
The tests will often be run as a non-root user, in which case the lvm cache would be empty. It would be better to use `mock` to fake the lvm data for your test.

Reply:
Fixed by mocking `lvs_info.cache` property.